### ISSUE TITLE
eq3btsmart

### ISSUE THREAD

**Issue by EuleMitKeule** [Heckie75/Eqiva-Smart-Radiator-Thermostat#1]
Moin!

Wenn ich gewusst hätte, dass zeitgleich mit mir noch jemand gerade eine Python-Library für die eQ-3-Thermostate schreibt, hätte ich vorgeschlagen, dass man sich zusammen tut. 😄
Ich habe nämlich gerade selbst ein größeres Update für meine eQ-3-Library fertiggestellt: https://github.com/EuleMitKeule/eq3btsmart
Falls dir die noch nicht bekannt ist, ich habe die hauptsächlich zur Nutzung in der aktuellen Home Assistant Core-Integration für die Thermostate gebaut, die ich ebenfalls maintaine. Eine CLI gibt es aktuell noch nicht, ist aber geplant. Ich hatte außerdem überlegt, mich mal dem Thema Firmware-Updates anzunehmen. Das sollte sich vermutlich anhand der dekompilierten App nachbilden lassen.

**Comment by Heckie75** [comment]
Hallo @EuleMitKeule ,

Tatsächlich wusste ich, dass es schon mehrere Implementierungen gibt. Ich habe vor ca. 7 Jahren die API des Eq3 analysiert und dann das erste CLI geschrieben und zwar: https://github.com/Heckie75/eQ-3-radiator-thermostat

Ich habe mehrere CLIs dieser Art für unterschiedliche BLE Geräte (siehe mein Repository). Viele dieser Skripte setzen  auf _gatttool_ , welches mit _expect_ ferngesteuert wird. Das war zur damaligen Zeit noch in Ordnung. Heute gibt es gute Bibliotheken, wie Bleak, die sogar Plattformunabhängig sind. Daher habe ich mir vorgenommen, die Skripte nach und nach zu ersetzen. Die Mipow Playbulbs habe ich schon migriert, siehe https://github.com/Heckie75/mipow-bulbs. Die Thermometer von Govee hatte ich von vorneherein mit Bleak implementiert, siehe https://github.com/Heckie75/govee-h5075-thermo-hygrometer

Als nächstes wären die BLE Smartmeter Voltcraft SEM-6000 und Voltcrsaft SEM-3600bt dran, sowie die Integration in Kodi. (Für SEM-6000 gibt es auch schon was, aber ich mache es dann trotzdem in meinem Stil und vor allem "feature complete")

Die Firmware-Updates hatte ich schon mal analysiert und angefangen. Ich habe hier auch noch Code umliegen, den ich offenbar nicht eingecheckt habe. Kann ich Dir gerne bereitstellen.  Da ich es nicht als sinnvoll erachte, Firmware updates außerhalb der App oder überhaupt zu machen, werde ich es nicht weiter verfolgen. Die Thermostate sind ja auch end-of-life, so dass mit Updates nicht mehr zu rechnen ist. Wie weit ich da gewesen bin, kann ich nicht sagen. Ist etwa 4 Jahre her.

**Comment by EuleMitKeule** [comment]
> aber ich mache es dann trotzdem in meinem Stil und vor allem "feature complete"

Ja, das kann ich absolut nachvollziehen. Habe aus ähnlichen Gründen auch lieber ein eigenes Projekt gestartet.

> Ich habe hier auch noch Code umliegen, den ich offenbar nicht eingecheckt habe. Kann ich Dir gerne bereitstellen.

Das wäre super! Vielleicht einfach als Gist oder auf pastebin? Dann könntest Du hier einfach einen Link posten und ich hätte schon mal einen Anhaltspunkt.

> Die Thermostate sind ja auch end-of-life, so dass mit Updates nicht mehr zu rechnen ist

Es gab tatsächlich kürzlich erst eine neue Firmwareversion 148, in der das Pairing per PIN gefixt wurde. Vorher war es wohl aufgrund eines Fehlers in der Firmware möglich, auch ohne PIN zu koppeln. Das geht jetzt nicht mehr.

**Comment by prefixFelix** [comment]
@Heckie75 Danke für die gute CLI und die ausführliche Protokollbeschreibung! 
Hab auf der Basis eine kleine [MicroPython library](https://github.com/prefixFelix/eqiva-radiator-thermostat) erstellt.
